sktree/stats/forestht.py

```diff
@@ -0,0 +1,466 @@
+
+    null_dist_ : ArrayLike of shape (n_repeats,)
+        The null distribution of the test statistic.
+    """
+
+    def __init__(
+        self,
+        n_estimators=100,
+        criterion="gini",
+        splitter="best",
+        max_depth=None,
+        min_samples_split=2,
+        min_samples_leaf=1,
+        min_weight_fraction_leaf=0.0,
+        max_features="sqrt",
+        max_leaf_nodes=None,
+        min_impurity_decrease=0.0,
+        bootstrap=False,
+        oob_score=False,
+        n_jobs=None,
+        random_state=None,
+        verbose=0,
+        warm_start=False,
+        class_weight=None,
+        ccp_alpha=0.0,
+        max_samples=None,
+        honest_prior="empirical",
+        honest_fraction=0.5,
+        tree_estimator=None,
+        alpha=0.05,
+    ):
+        self.estimator = HonestForestClassifier(
+            n_estimators=n_estimators,
+            criterion=criterion,
+            splitter=splitter,
+            max_depth=max_depth,
+            min_samples_split=min_samples_split,
+            min_samples_leaf=min_samples_leaf,
+            min_weight_fraction_leaf=min_weight_fraction_leaf,
+            max_features=max_features,
+            max_leaf_nodes=max_leaf_nodes,
+            min_impurity_decrease=min_impurity_decrease,
+            bootstrap=bootstrap,
+            oob_score=oob_score,
+            n_jobs=n_jobs,
+            random_state=random_state,
+            verbose=verbose,
+            warm_start=warm_start,
+            class_weight=class_weight,
+            ccp_alpha=ccp_alpha,
+            max_samples=max_samples,
+            honest_prior=honest_prior,
+            honest_fraction=honest_fraction,
+            tree_estimator=tree_estimator,
+        )
+        self.alpha = alpha
+        self.n_jobs = n_jobs
+        self.n_estimators = n_estimators
+        self.criterion = criterion
+        self.splitter = splitter
+        self.max_depth = max_depth
+        self.min_samples_split = min_samples_split
+        self.min_samples_leaf = min_samples_leaf
+        self.min_weight_fraction_leaf = min_weight_fraction_leaf
+        self.max_features = max_features
+        self.max_leaf_nodes = max_leaf_nodes
+        self.min_impurity_decrease = min_impurity_decrease
+        self.bootstrap = bootstrap
+        self.oob_score = oob_score
+        self.random_state = random_state
+        self.verbose = verbose
+        self.warm_start = warm_start
+        self.class_weight = class_weight
+        self.ccp_alpha = ccp_alpha
+        self.max_samples = max_samples
+        self.honest_prior = honest_prior
+        self.honest_fraction = honest_fraction
+        self.tree_estimator = tree_estimator
+
+    def _statistic(
+        self,
+        estimator: ForestClassifier,
+        X: ArrayLike,
+        y: ArrayLike,
+        covariate_index: ArrayLike = None,
+        metric="auc",
+        test_size=0.2,
+        **metric_kwargs,
+    ):
```

Comment:
I think it will still be a good optional functionality to output the posteriors, so we can save time on potential metric modifications in the future.

Reply:
Added